lib/Runtime/Library/EngineInterfaceObject.cpp

```diff
@@ -315,32 +315,47 @@ namespace Js
 
     Var EngineInterfaceObject::Entry_TagPublicLibraryCode(RecyclableObject *function, CallInfo callInfo, ...)
     {
+#pragma warning(push)
+#pragma warning(suppress: 4189) // scriptContext is initialized by never used
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+#pragma warning(pop)
 
-        if (callInfo.Count >= 2 && JavascriptFunction::Is(args.Values[1]))
+        AssertOrFailFast((callInfo.Count == 3 || callInfo.Count == 4) && JavascriptFunction::Is(args[1]) && JavascriptString::Is(args[2]));
+
+        JavascriptFunction *func = JavascriptFunction::UnsafeFromVar(args[1]);
+        JavascriptString *methodName = JavascriptString::UnsafeFromVar(args[2]);
+
+        func->GetFunctionProxy()->SetIsPublicLibraryCode();
+
+        const char16 *methodNameBuf = methodName->GetSz();
```

Comment:
I don't expect the intl js code to do anything too zany here, but is it worth using `GetString` instead of `GetSz`? Since we're passing the length around anyway either should work, and I know that @sethbrenith  has made some good points that it is typically better hygiene to use the `GetString` version

Reply:
We use `wcsrchr` below, which doesn't take a length parameter. It used to be GetString and we always call it on literal strings, but in case we ever call it on a non-literal-string in the future, the `wcsrchr` would start searching past the end of the string we actually wanted.

Reply:
Makes sense.

Reply:
Also, this runs on Intl initialization, so its not particularly hot. I can add a comment for it.